lyrics_audio_embeddings.py

```diff
@@ -6,100 +6,116 @@
+            embedding = audio_transformer(waveform)
+            return embedding.cpu().numpy()
+    except Exception as e:
+        print(f"Error processing {file_path}: {str(e)}")
+        return None
+
+def process_and_save_audio_embeddings(input_folder, output_file):
+    """Process all audio files and save embeddings"""
+    device = setup_audio_backend()
+    audio_transformer.to(device)
+    audio_transformer.eval()
+
+    embeddings = []
+    song_ids = []
+    mp3_files = [f for f in os.listdir(input_folder) if f.endswith('.mp3')]
+
+    for idx, file_name in enumerate(mp3_files, 1):
+        file_path = os.path.join(input_folder, file_name)
+        embedding = process_audio(file_path, device)
+        if embedding is not None:
+            embeddings.append(embedding)
+            song_ids.append(file_name.replace('.mp3', ''))
+            print(f"Successfully processed {file_name}")
+        else:
+            print(f"Failed to process {file_name}")
+
+    os.makedirs(os.path.dirname(output_file), exist_ok=True)
+    np.savez(output_file, song_ids=song_ids, embeddings=embeddings)
+    print(f"Audio embeddings saved to {output_file}")
 
 def process_and_save_lyrics_embeddings(input_folder, output_file):
-    """
-    Process all lyrics in a folder, extract embeddings, and save them to a file.
-    Args:
-        input_folder (str): Path to folder containing cleaned lyric files.
-        output_file (str): Path to save embeddings.
-    """
+    """Process all lyrics files and save embeddings"""
     embeddings = []
     song_ids = []
 
     for file_name in os.listdir(input_folder):
         file_path = os.path.join(input_folder, file_name)
-        if os.path.isfile(file_path) and file_path.endswith('.txt'):
-            # Read the cleaned lyrics
+        if file_path.endswith('.txt'):
             with open(file_path, 'r', encoding='utf-8') as file:
                 lyrics = file.read()
 
-            # Tokenize and split into chunks of max length 256
             tokens = tokenizer(
                 lyrics,
                 return_tensors="pt",
                 truncation=True,
                 max_length=256,
-                padding=False  # No padding for chunks
+                padding=False
             )
+            input_ids_chunks = torch.split(tokens["input_ids"], 256, dim=1)
 
-            input_ids_chunks = tokens["input_ids"].split(dim=1, split_size=256)  # Split into 256-token chunks
-
-            # Generate embedding for each chunk and average them
             chunk_embeddings = []
             with torch.no_grad():
                 for chunk in input_ids_chunks:
                     chunk_embedding = text_transformer(chunk)
                     chunk_embeddings.append(chunk_embedding)
 
-            # Average embeddings across chunks
             aggregated_embedding = torch.mean(torch.stack(chunk_embeddings), dim=0)
-            embeddings.append(aggregated_embedding.detach().numpy())
```

Comment:
we need detach to make a copy
If we are using it as a separate variable. 

Else it will go into the gradient calculations.